### PR TITLE
Server Capabilities

### PR DESCRIPTION
**Pull request by Hoikas**
Implements server capabilities broadcast as discussed for #112. This requires H-uru/Plasma#546

### CODE REVIEW

File: AuthServ/AuthServer.cpp [hunk 1]
```diff
@@ -77,8 +77,14 @@ void auth_init(AuthServer_Private& client)
         client.m_buffer.writeBytes(serverSeed, 7);
     }
 
-    /* send reply */
+    /* send encryption reply */
     DS::SendBuffer(client.m_sock, client.m_buffer.buffer(), client.m_buffer.size());
+
+    /* inform client of server caps as part of the init process */
+    START_REPLY(e_AuthToCli_ServerCaps);
+    uint32_t caps = e_CapsScoreLeaderBoards;
```
Comment (Hoikas) [inline AuthServ/AuthServer.cpp]: The reason this value is stored as a temporary is for future configurable capabilities, such as the planned cap for the Gahreesen wall.

**Comment by zrax** [comment]
Do you think 32 bits will be enough for everything we may want to add in the future?  I know the client makes varying length fields a bit more difficult, but I feel like we should reserve at least one more dword in case we come up with something that needs more than just a boolean bit.

In addition (more a note to others who come here since I'm sure @Hoikas is already aware):  This method of setting caps will BOOT any client that doesn't have support for the message.  If you run a server with this change, you MUST also be using a client that has H-uru/Plasma#546.

**Comment by Hoikas** [comment]
I was operating under the assumption that anyone using this changeset would be operating in lockstep with H-uru/Plasma master and that we could add new bits as needed... I could up this to a 64-bit field. I bet eap's netcode doesn't support that natively, but new field types are fairly trivial to add.

**Comment by zrax** [comment]
I was thinking more along the lines of adding a second DWORD.  But your point about keeping this in sync with the client is valid, so I'm fine with leaving it as a single value.

**Comment by dpogue** [comment]
I guess we don't have the option of just sticking a hsBitVector in there?

**Comment by Hoikas** [comment]
We could encode an hsBitVector inside a payload instead of sending down a DWORD, it's just a bit more work. Thoughts, @zrax?

**Comment by zrax** [comment]
Using a bit vector makes it easier to adjust in the future, so if it's not too painful to add, I think I like the option better.

**Comment by Hoikas** [comment]
Over a year later this has been updated to use an hsBitVector as requested 😛

**Comment by Hoikas** [comment]
:boom:

**Pull request merged**